client/signup/main.jsx

```diff
@@ -460,7 +460,9 @@ class Signup extends React.Component {
 				( this.props.signupDependencies &&
 					this.props.signupDependencies.domainItem &&
 					this.props.signupDependencies.domainItem.is_domain_registration &&
```

Comment:
I believe we set the `domainItem` in `signupDependencies` too. Can probably do:

( this.props.signupDependencies.domainItem.is_domain_registration || this.props.signupDependencies.domainItem.product_slug === 'domain_transfer' )